src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,31 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(boolean isPrivate) {
+        this.isPrivate = isPrivate;
```

Comment:
Could you have extracted more into the superclass? Think about what each of the subclasses constructor does in common.